d2l/tensorflow.py

```diff
@@ -1847,7 +1851,7 @@ def train_seq2seq(net, data_iter, lr, num_epochs, tgt_vocab, device):
         if (epoch + 1) % 10 == 0:
             animator.add(epoch + 1, (metric[0] / metric[1],))
     print(f'loss {metric[0] / metric[1]:.3f}, {metric[1] / timer.stop():.1f} '
-          f'tokens/sec on {str(device)}')
+          f'tokens/sec on {str(device._device_name)}')
```

Comment:
Please make the same change in https://github.com/d2l-ai/d2l-en/blob/master/chapter_appendix-tools-for-deep-learning/utils.md and then run d2lbook build lib. We don't directly change the lib files, these are automatically generated using the d2lbook package after editing the source files where these functions are saved.